test/tangent_types/abstract_zero.jl

```diff
@@ -154,3 +154,16 @@
         @test isempty(detect_ambiguities(M))
     end
 end
+
+@testset "zero_tangent" begin
```

Comment:
Please consider adding test cases for a few more types with immutable tangents (primitive types like `Float64`, `Int`, and a `struct` of some kind.